apps/web/hooks/useS3.tsx

```diff
@@ -69,6 +69,8 @@ export const S3Provider: React.FC<PropsWithChildren<Props>> = ({
 			data.keys.bucketUrl = `https://${data.keys.Bucket}.s3.${data.keys.region}.backblazeb2.com`;
 		} else if ((Provider[data.type] as Provider) === Provider.wasabi) {
 			data.keys.bucketUrl = `https://${data.keys.Bucket}.s3.${data.keys.region}.wasabisys.com`;
+		} else if ((Provider[data.type] as Provider) === Provider.digitalocean) {
+			data.keys.bucketUrl = `https://${data.keys.Bucket}.s3.${data.keys.region}.digitalocean.com`;
```

Comment:
Same here; we can create a helper function that allows us to get the bucket url:
```
const getBucketUrl = (data) => {
  const provider = Provider[data.type];

  switch (provider) {
    case Provider.s3:
      return `https://${data.keys.Bucket}.s3.${data.keys.region}.amazonaws.com`;
    case Provider.backblaze:
      return `https://${data.keys.Bucket}.s3.${data.keys.region}.backblazeb2.com`;
    case Provider.wasabi:
      return `https://${data.keys.Bucket}.s3.${data.keys.region}.wasabisys.com`;
    case Provider.digitalocean:
      return `https://${data.keys.Bucket}.s3.${data.keys.region}.digitalocean.com`;
    default:
      return null; // or throw an error if necessary
  }
};

data.keys.bucketUrl = getBucketUrl(data);
```

We apply separation of concerns by having one line handle assignment and one function determine the bucket URL.

Using a switch statement is easy to expand upon as well.